Include/internal/pycore_genobject.h

```diff
@@ -22,6 +22,13 @@ extern "C" {
     PyObject *prefix##_qualname;                                            \
     _PyErr_StackItem prefix##_exc_state;                                    \
     PyObject *prefix##_origin_or_finalizer;                                 \
+    /* A *borrowed* reference to a task that drives the coroutine.          \
```

Comment:
Also, I don't see how it is safe for this to be a borrowed reference. What guarantee is there that the object passed to `_PyCoro_SetTask` will outlive the generator?

If the answer is "_PyCoro_SetTask is an internal function", then why is it used by asyncio which is a module?

Reply:
> This is adding yet another field generators for the use of async.

AFAICT there is too much entanglement between generators and coroutines to split their implementation entirely to avoid this. How can we measure what this unwanted expense actually is?

Reply:
Splitting their implementations is desirable, but I appreciate that it won't happen in this PR.

Reply:
`cr_task` is a borrowed reference, and is exposed through a public API, which is unsafe.
So it will need to made a strong reference.

That means that it not only does it takes space and needs initializing, it will also need to be cleared when the generator is destroyed.

Reply:
@markshannon we agree with your feedback. I'll refactor the PR to remove the need for `cr_task` (and therefore the `gi_task` debug offset). We'll maintain a list of running tasks in _asynciomodule.c since this is asyncio-specific anyway.